fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -2459,27 +2676,108 @@ private TableName qualifiedNameToTableName(QualifiedName qualifiedName) {
         }
     }
 
-    private Type getType(StarRocksParser.TypeContext type) {
-        Integer length = null;
+    public Type getType(StarRocksParser.TypeContext context) {
+        if (context.baseType() != null) {
+            return getBaseType(context.baseType());
+        } else if (context.decimalType() != null) {
+            return getDecimalType(context.decimalType());
+        } else if (context.arrayType() != null) {
+            return getArrayType(context.arrayType());
+        }
+        throw new IllegalArgumentException("Unsupported type specification: " + context.getText());
+    }
+
+    private Type getBaseType(StarRocksParser.BaseTypeContext context) {
+        int length = -1;
+        if (context.typeParameter() != null) {
+            length = Integer.parseInt(context.typeParameter().INTEGER_VALUE().toString());
+        }
+        if (context.STRING() != null) {
+            ScalarType type = ScalarType.createVarcharType(ScalarType.DEFAULT_STRING_LENGTH);
+            type.setAssignedStrLenInColDefinition();
+            return type;
+        } else if (context.VARCHAR() != null) {
+            ScalarType type = ScalarType.createVarcharType(length);
+            if (length != -1) {
+                type.setAssignedStrLenInColDefinition();
+            }
+            return type;
+        } else if (context.CHAR() != null) {
+            ScalarType type = ScalarType.createCharType(length);
+            if (length != -1) {
+                type.setAssignedStrLenInColDefinition();
+            }
+            return type;
+        } else if (context.SIGNED() != null) {
+            return Type.INT;
+        } else if (context.HLL() != null) {
+            ScalarType type = ScalarType.createHllType();
+            type.setAssignedStrLenInColDefinition();
+            return type;
+        } else {
+            return ScalarType.createType(context.getChild(0).getText());
+        }
+    }
+
+    public ScalarType getDecimalType(StarRocksParser.DecimalTypeContext context) {
         Integer precision = null;
         Integer scale = null;
-
-        if (type.baseType() != null) {
-            if (type.baseType().typeParameter() != null) {
-                length = Integer.parseInt(type.baseType().typeParameter().INTEGER_VALUE().toString());
+        if (context.precision != null) {
+            precision = Integer.parseInt(context.precision.getText());
+            if (context.scale != null) {
+                scale = Integer.parseInt(context.scale.getText());
+            }
+        }
+        if (context.DECIMAL() != null) {
+            if (precision != null) {
+                if (scale != null) {
+                    return ScalarType.createUnifiedDecimalType(precision, scale);
+                }
+                try {
+                    return ScalarType.createUnifiedDecimalType(precision);
+                } catch (AnalysisException e) {
```

Comment:
```suggestion
```

Reply:
An  AnalysisException  exception is thrown here, can not delete try catch code